pkg/engine/task/task.go

```diff
@@ -15,6 +16,7 @@ import (
 // Context is a engine.task execution context containing k8s client, templates parameters etc.
 type Context struct {
 	Client     client.Client
+	Config     *rest.Config
```

Comment:
Should this be a `rest.Config` and not a pointer? I'm not too sure about the go semantics when to use a pointer and when not, but a pointer indicates that the var can be modified?

Not a blocker for me though.

Reply:
I don't think using a pointer or not has any impact on mutability.
`rest.Config` is a struct, so one uses a pointer when you don't want to make a copy.
This seems consistent with how I see this object type being used elsewhere.

Reply:
The struct member can always be modified here so this isn't a blocker. This is correct. For embedded structs, typically it's better to copy the ref rather than copy the entire struct (with exceptions). @ANeumann82 as a note for other code as well, interfaces are ALWAYS written as the interface value (not with a `*` for the pointer), but are ALWAYS a reference.